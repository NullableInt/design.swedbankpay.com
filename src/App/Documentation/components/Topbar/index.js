import React, { Component } from "react";
import PrismCode from "react-prism";

import { ComponentPreview, DocToc, Property, Attribute } from "#";
import TopbarComponent from "@/Topbar";
import { topbar } from "$/px-script/main";
import Alert from "@/Alert";

const menu = {
    btn: {
        icon: "menu",
        text: "Menu"
    },
    items: ["Link 1", "Link 2", "Link 3"]
};

const topbarContent = {
    id: "topbar-nav",
    ...menu
};

// TODO: Remove when deprecated parts are deleted.
const DeprecatedAlert = () => (
    <Alert type="danger">
        <h3>Component contains deprecated parts</h3>
        <p>The following parts are deprecated and will be removed:</p>
        <ul>
            <li>Right nav</li>
            <li>Levels in left nav</li>
            <li>topbar-btn-right and topbar-btn-left classes will be replaced by topbar-btn</li>
        </ul>
        <p>For continued support, please follow the examples in the documentation.</p>
    </Alert>
);

const Overview = () => (
    <>
        <h2 id="overview">Overview</h2>
        <p>
            To use a topbar the bare minimum markup required is:
        </p>
        <ul>
            <li><Property value=".topbar" /> wrapper.</li>
            <li>Anchor with class <Property value=".topbar-logo" /> to display the logo and make it clickable.</li>
        </ul>
        <p>
            With this you get a topbar with the PayEx logo.
            In the topbar you can add additional functionality like a menu containing links. To enable this add a <PrismCode className="language-html">{"<button>"}</PrismCode> and
            a <PrismCode className="language-html">{"<nav>"}</PrismCode> containing anchors. Remember to include <Attribute data name="toggle-nav" value="{your_nav_id}"/> as a <PrismCode className="language-html">{"<button>"}</PrismCode> attribute
            to get the menu to appear when the menu-button is clicked.
        </p>
        <ComponentPreview language="html" showCasePanel codeFigure>
            <TopbarComponent topbarContent={topbarContent} logout />
        </ComponentPreview>
    </>
);

const FixedTopbar = () => (
    <>
        <h2 id="fixed-topbar">Fixed topbar</h2>
        <p>
            Add <Property value=".topbar-fixed" /> with your <Property value=".topbar" /> to get a fixed topbar.
            For a demonstration of the usage see the documentations top bar or implement it in your own project!
        </p>
        <ComponentPreview language="html" codeFigure>
            <TopbarComponent fixed />
        </ComponentPreview>
    </>
);

const TopbarText = () => (
    <div className="col-lg-10 doc-body">
        <DeprecatedAlert />
        <p className="lead">Topbar...</p>
        <Overview />
        <FixedTopbar />
    </div>
);

class Topbar extends Component {
    componentDidMount () {
        topbar.init(1);
    }

    render () {
        return (
            <div className="doc-container">
                <div className="row">
                    <TopbarText />
                    <DocToc component={TopbarText} />
                </div>
            </div>
        );
    }
}

export default Topbar;

/* For testing */
export { Overview, TopbarText, FixedTopbar, DeprecatedAlert };
