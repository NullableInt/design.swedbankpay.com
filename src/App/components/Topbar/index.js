import React, { Fragment } from "react";
import PropTypes from "prop-types";

const TopbarBtn = ({ align, icon, text, target }) => (
    <button type="button" className={`topbar-btn-${align}`} data-toggle-nav={target ? `#${target}` : null}>{"\n\t\t"}
        {icon ?
            <>
                <i className="material-icons topbar-btn-icon">{icon}</i>
                {"\n\t\t"}
            </>
            : null}
        {text ?
            <>
                <span className="topbar-btn-text">{text}</span>
                {"\n\t"}
            </>
            : null}
    </button>
);

const TopbarMenu = ({ menu }) => {
    const { id, items } = menu;

    return (
        <nav id={id} className={"topbar-nav"}>
            {items.map((item, i) => (
                <Fragment key={i}>
                    {"\n"}<a href="#" onClick={e => e.preventDefault()}>{item}</a>
                </Fragment>
            ))}{"\n"}
        </nav>
    );
};

const Topbar = ({ topbarContent, fixed, logout }) => (
    <header className={`topbar ${fixed ? "topbar-fixed" : ""}`}>{"\n"}
        {topbarContent ?
        <>
            {topbarContent.btnLeft ?
                <>
                    <TopbarBtn align="left" icon={topbarContent.btnLeft.icon} text={topbarContent.btnLeft.text} target={topbarContent.id} />{"\n"}
                    <TopbarMenu menu={topbarContent} />
                </> :
                null}
            {"\n"}<a href="#" className={"topbar-logo"} onClick={e => e.preventDefault()}></a>{"\n"}
            {logout ?
                <>
                    <TopbarBtn align="right" icon="exit_to_app" text="Log out" />{"\n"}
                </> :
                null}
        </>
            :
        <>
            <a href="#" className={"topbar-logo"} onClick={e => e.preventDefault()}></a>{"\n"}
        </>}
    </header>
);

Topbar.propTypes = {
    topbarContent: PropTypes.object,
    fixed: PropTypes.bool,
    logout: PropTypes.bool
};

export default Topbar;
