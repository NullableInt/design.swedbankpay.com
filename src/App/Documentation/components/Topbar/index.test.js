import React from "react";
import { shallow } from "enzyme";

import Topbar, { Overview, TopbarText } from "./index";

describe("Components: Topbar", () => {
    it("is defined", () => {
        expect(Topbar).toBeDefined();
    });

    it("renders", () => {
        const wrapper = shallow(<Topbar />);

        expect(wrapper).toMatchSnapshot();
    });

    describe("Overview", () => {
        it("is defined", () => {
            expect(Overview).toBeDefined();
        });

        it("renders", () => {
            const wrapper = shallow(<Overview />);

            expect(wrapper).toMatchSnapshot();
        });
    });

    describe("TopbarText", () => {
        it("is defined", () => {
            expect(TopbarText).toBeDefined();
        });

        it("renders", () => {
            const wrapper = shallow(<TopbarText />);

            expect(wrapper).toMatchSnapshot();
        });
    });
});
