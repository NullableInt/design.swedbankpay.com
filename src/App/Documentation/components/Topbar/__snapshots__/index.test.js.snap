// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components: Topbar DeprecatedAlert renders 1`] = `
<Alert
  type="danger"
>
  <h3>
    Component contains deprecated parts
  </h3>
  <p>
    The following parts are deprecated and will be removed:
  </p>
  <ul>
    <li>
      Right nav
    </li>
    <li>
      Levels in left nav
    </li>
  </ul>
  <p>
    For continued support, please follow the examples in the documentation.
  </p>
</Alert>
`;

exports[`Components: Topbar FixedTopbar renders 1`] = `
<Fragment>
  <h2
    id="fixed-topbar"
  >
    Fixed topbar
  </h2>
  <p>
    Add 
    <Property
      value=".topbar-fixed"
    />
     with your 
    <Property
      value=".topbar"
    />
     to get a fixed topbar. For a demonstration of the usage see the documentations top bar or implement it in your own project!
  </p>
  <ComponentPreview
    codeFigure={true}
    language="html"
  >
    <Topbar
      fixed={true}
    />
  </ComponentPreview>
</Fragment>
`;

exports[`Components: Topbar Overview renders 1`] = `
<Fragment>
  <h2
    id="overview"
  >
    Overview
  </h2>
  <p>
    To use a topbar the bare minimum markup required is:
  </p>
  <ul>
    <li>
      <Property
        value=".topbar"
      />
       wrapper.
    </li>
    <li>
      Anchor with class 
      <Property
        value=".topbar-logo"
      />
       to display the logo and make it clickable.
    </li>
  </ul>
  <p>
    With this you get a topbar with the PayEx logo. In the topbar you can add additional functionality like a menu containing links. To enable this add a 
    <PrismCode
      className="language-html"
      component="code"
    >
      &lt;button&gt;
    </PrismCode>
     and a 
    <PrismCode
      className="language-html"
      component="code"
    >
      &lt;nav&gt;
    </PrismCode>
     containing anchors. Remember to include 
    <Attribute
      data={true}
      name="toggle-nav"
      value="{your_nav_id}"
    />
     as a 
    <PrismCode
      className="language-html"
      component="code"
    >
      &lt;button&gt;
    </PrismCode>
     attribute to get the menu to appear when the menu-button is clicked.
  </p>
  <ComponentPreview
    codeFigure={true}
    language="html"
    showCasePanel={true}
  >
    <Topbar
      topbarContent={
        Object {
          "btnLeft": Object {
            "icon": "menu",
            "text": "Menu",
          },
          "btnRight": Object {
            "icon": "exit_to_app",
            "text": "Log out",
          },
          "id": "topbar-nav",
          "items": Array [
            Object {
              "href": "#",
              "title": "Link 1",
            },
            Object {
              "href": "#",
              "title": "Link 2",
            },
            Object {
              "href": "#",
              "title": "Link 3",
            },
          ],
        }
      }
    />
  </ComponentPreview>
</Fragment>
`;

exports[`Components: Topbar TopbarText renders 1`] = `
<div
  className="col-lg-10 doc-body"
>
  <DeprecatedAlert />
  <p
    className="lead"
  >
    Topbar...
  </p>
  <Overview />
  <FixedTopbar />
</div>
`;

exports[`Components: Topbar renders 1`] = `
<div
  className="doc-container"
>
  <div
    className="row"
  >
    <TopbarText />
    <DocToc
      component={[Function]}
    />
  </div>
</div>
`;
